src/arch/x86_64/fdt.rs

```diff
@@ -0,0 +1,50 @@
+extern crate alloc;
```

Comment:
`extern crate` declarations should go into the crate root.

Reply:
I tried to move `extern crate alloc` into `main.rs`. But `clippy` warned about unused import, even when I add `#[cfg(all(target_os = "none", not(feature = "fc")))]` before.

Reply:
That's because this is not yet used on AArch64 and RISC-V. This should work:

```rust
#[cfg(all(target_arch = "x86_64", target_os = "none", not(feature = "fc")))]
extern crate alloc;
```